src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -1019,4 +1019,30 @@ public function isAtLeastOneLocalPrimaryKey()
 
         return 0 !== count($cols);
     }
+
+    public function isTheSamePolymorphicSignature(ForeignKey $fk)
+    {
+        return $this->getPolymorphicSignature() == $fk->getPolymorphicSignature();
+    }
+
+    public function isAPolymorphicSiblingRelationship(ForeignKey $fk)
+    {
+        if (!($this->isPolymorphic() && $fk->isPolymorphic())) {
+            return false;
+        }
+
+        return $this->isTheSamePolymorphicSignature($fk);
+    }
+
+    public function getPolymorphicSignature()
+    {
+        if (!$this->isPolymorphic()) {
+            return false;
+        }
+
+        $sortedCols = $this->getLocalColumns();
+        sort($sortedCols);
+
+        return sha1(implode('|||', $sortedCols));
```

Comment:
I don't know about everyone else, but any time I've wanted to generate something like a "unique signature" it's always come back to haunt me in 2 ways:

1. It ends up being slow when called hundreds of times in a loop.
2. It ends up being really hard to change in a BC way because it isn't abstracted out of the class itself.

It might be a good feature but we should look at improving these two points. Perhaps the signature generation could be even simpler than it is, and/or it could also be moved out of the class.